### ISSUE TITLE
Code generated by protoc 3.18.0 not compatible with protobuf-java 3.17.3 (and earlier)

### ISSUE THREAD

**Issue by augi** [protocolbuffers/protobuf#8985]
**What version of protobuf and what language are you using?**
Version: 3.18.0 (both protoc and protobuf-java)
Language: Java

**What operating system (Linux, Windows, ...) and version?**
Windows, Linux

**What runtime / compiler are you using (e.g., python version or gcc version)**
Gradle Protobuf plugin 0.8.17

**What did you do?**
After updating Protobuf from `3.17.3` to `3.18.0`, we can see the following error:
```
D:\Work\public\rabbitmq-scala-client\extras-scalapb\build\generated\source\proto\test\java\com\avast\clients\rabbitmq\test\ExampleEvents.java:226: error: cannot find symbol
      if (!com.google.protobuf.GeneratedMessageV3.isStringEmpty(fileId_)) {
                                                 ^
  symbol:   method isStringEmpty(java.lang.Object)
  location: class com.google.protobuf.GeneratedMessageV3
```

Looks like the generated code tries to call `com.google.protobuf.GeneratedMessageV3.isStringEmpty` method that `protected` and so not accessible here. It fails the compilation.

Pull request updating Protobuf to 3.18.0, and thus reproducing this bug: https://github.com/avast/rabbitmq-scala-client/pull/140

**What did you expect to see**
No compilation error of the generated code.

**What did you see instead?**
Compilation error.

**Anything else we should know about your project / environment**

**Comment by perezd** [comment]
This PR did it:
https://github.com/protocolbuffers/protobuf/pull/7526

**Comment by elharo** [comment]
I'm surprised no test caught this. Do we not have any tests that compile the generated Java code?

**Comment by perezd** [comment]
yes, nearly all the tests compile protos AFAIK.

**Comment by elharo** [comment]
I think we should just rollback #7526 for now.

**Comment by perezd** [comment]
SGTM

**Comment by elharo** [comment]
Though technically that's an API breaking change

**Comment by perezd** [comment]
making it public is probably the way forward here then.

**Comment by elharo** [comment]
I think we can revert the change in protoc and leave the change in GeneratedMessageV3.java for now. That avoids changing the published API.

**Comment by elharo** [comment]
This is the proto file where the compile fails:

https://github.com/avast/rabbitmq-scala-client/blob/master/core/src/test/proto/ExampleEvents.proto

**Comment by perezd** [comment]
Can you confirm you're using the 3.18.0 runtime as well as protoc?

**Comment by perezd** [comment]
Can you also verify the gradle plugin isn't providing an older runtime that might be getting picked up in the classpath?

**Comment by augi** [comment]
Hello, we have these snippets in the Gradle project:
```
classpath 'com.google.protobuf:protobuf-gradle-plugin:0.8.17'

protobufVersion = "3.18.0"

api "com.google.protobuf:protobuf-java:$protobufVersion"

protobuf {
    protoc {
        artifact = "com.google.protobuf:protoc:$protobufVersion"
    }
}
```
Also IDEA claim `3.18.0` dependency is used.
![image](https://user-images.githubusercontent.com/273052/133566852-b3b1216b-e6ef-467b-86bf-bf5b2230804b.png)

I've also tried to execute the downloaded `protoc` and its version seems to be correct:
```
C:\Users\augustyn\.gradle\caches\modules-2\files-2.1\com.google.protobuf\protoc\3.18.0\1641536d5a8bbfca63aad8b5284b213c0fb2ee1>protoc-3.18.0-windows-x86_64.exe --version
libprotoc 3.18.0
```

**Comment by augi** [comment]
OK, I've just found the root cause, and I'm very sorry. I got lost in the `rabbitmq-scala-client` project - there are two subprojects:
* `extras-protobuf` - it works well
* `extras-scalapb` - this is the problematic one. The issue is that we don't define `protobuf-java` dependency explicitly, but we are getting this dependency via `scalapb`, and it comes with `3.15.8`.
* 
So the issue actually is that there is breaking change between `3.18.0` and earlier `3.x` version. The code generated by `protoc` `3.18.0` don't compile with earlier `3.x` `protobuf-java` versions. I will update the summary of this issue accordingly.
